### PR TITLE
Delegate API

### PR DESCRIPTION
**Pull request by nathanielparke**
PR Builds out the basics for a delegate creation and list API

### CODE REVIEW

**Comment by nathanielparke** [comment]
Still have to build out tests but after that this API should be a good starting point for delegates

File: huxley/api/tests/test_delegate.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            "summary" : unicode(self.delegate.summary)}
+        )
+
+
+class DelegateDetailPutTestCase(UpdateAPITestCase):
+    url_name = 'api:delegate_detail'
+    params = {
+        'name':'Trevor Dowds',
+        'email':'tdowds@hotmail.org',
+        'summary':'He did awful!'}
```
Comment (nathanielparke) [inline huxley/api/tests/test_delegate.py]: @tdowds

Comment (tdowds) [inline huxley/api/tests/test_delegate.py]: Throwback to Delegate Summaries maybe?

File: huxley/accounts/models.py [hunk 2]
```diff
@@ -15,19 +15,24 @@ class User(AbstractUser):
 
     TYPE_ADVISOR = 1
     TYPE_CHAIR = 2
+    TYPE_DELEGATE = 3
     USER_TYPE_CHOICES = ((TYPE_ADVISOR, 'Advisor'),
-                         (TYPE_CHAIR, 'Chair'))
+                         (TYPE_CHAIR, 'Chair'),
+                         (TYPE_DELEGATE, 'Delegate'))
```
Comment (tdowds) [inline huxley/accounts/models.py]: Is this in anticipation that delegates will be able to log onto Huxley?

Comment (nathanielparke) [inline huxley/accounts/models.py]: Yes! I think we need to discuss the model framework I am thinking about for the Delegate users.

File: huxley/utils/test.py [hunk 3]
```diff
@@ -96,11 +96,38 @@ def new_committee(**kwargs):
...
+                name=kwargs.pop('name', 'TestCountry'),
+                special=kwargs.pop('special', False))
+        c.save()
+        return c
+
+class TestDelegates():
+    @staticmethod
+    def new_delegate(**kwargs):
+        a = TestAssignments.new_assignment()
+        c = Delegate(
```
Comment (tdowds) [inline huxley/utils/test.py]: Might make more sense to assign this to a variable called `d`. Just for consistency

**Comment by nathanielparke** [comment]
Merged in e1facecf98780358a12fa5ea22eb8e09ed02a1c0

**Pull request closed without merging**